File: funcscript/parser/syntax/FuncScriptParser.GetSpaceSepratedListExpression.cs
using funcscript.block;
using funcscript.model;

namespace funcscript.core
{
    public partial class FuncScriptParser
    {

        static ParseResult GetSpaceSepratedListExpression(ParseContext context, int index)
        {
            ParseNode parseNode = null;
            ListExpression listExpr = null;
            var i = SkipSpace(context, index).NextIndex;
            var tokenStart = i;
            var listItems = new List<ExpressionBlock>();
            var nodeListItems = new List<ParseNode>();
            (var firstItem,var nodeFirstItem, var i2) = GetExpression(context, i);
            if (i2 > i)
            {
                listItems.Add(firstItem);
                nodeListItems.Add(nodeFirstItem);
                i = i2;
                do
                {
                    i2 = GetLiteralMatch(context, i, new[] { " " }).NextIndex;
                    if (i2 == i)
                        break;
                    i = i2;
                    i = SkipSpace(context, i).NextIndex;
                    (var otherItem,var nodeOtherItem, i2) = GetExpression(context, i);
                    if (i2 == i)
                        break;
                    listItems.Add(otherItem);
                    nodeListItems.Add(nodeOtherItem);
                    i = i2;
                } while (true);
            }

            listExpr = new ListExpression { ValueExpressions = listItems.ToArray() };
            listExpr.SetContext(context.ReferenceProvider);
            parseNode = new ParseNode(ParseNodeType.List, index, i - tokenStart, nodeListItems);
            return new ExpressionBlockResult(listExpr, parseNode, i);
        }
    }
}

File: ts/core/src/parser/syntax/FuncScriptParser.GetSpaceSepratedListExpression.ts
import { ParseContext, ParseNode, ParseNodeType, ParseResult, SyntaxErrorData } from "../FuncScriptParser.Main";
import { GetExpression } from "./FuncScriptParser.GetExpression";
import { GetLiteralMatch } from "./FuncScriptParser.GetLiteralMatch";
import { SkipSpace } from "./FuncScriptParser.SkipSpace";

function GetSpaceSepratedListExpression(context: ParseContext, index: number): ParseResult {
    let parseNode: ParseNode | null = null;
    let i = SkipSpace(context, index).NextIndex;
    const tokenStart = i;
    const nodeListItems: ParseNode[] = [];
    let nodeFirstItem: ParseNode | null = null;
    let i2: number;

    // Use parentheses for destructuring and ensure variables can be reassigned
    ({ ParseNode: nodeFirstItem, NextIndex: i2 } = GetExpression(context, i));

    if (i2 > i) {
        nodeListItems.push(nodeFirstItem!);
        i = i2;
        do {
            i2 = GetLiteralMatch(context, i, " ").NextIndex;
            if (i2 === i) break;
            i = i2;
            i = SkipSpace(context, i).NextIndex;

            // Destructure properly or adjust depending on the function's return
            const { ParseNode: nodeOtherItem, NextIndex: newIndex } = GetExpression(context, i);
            if (newIndex === i) break;
            nodeListItems.push(nodeOtherItem!);
            i = newIndex;
        } while (true);
    }

    parseNode = new ParseNode(ParseNodeType.List, index, i - tokenStart, nodeListItems);
    return { ParseNode: parseNode, NextIndex: i };
}